runtime/gc_include/ObjectAllocationAPI.hpp

```diff
@@ -141,17 +142,25 @@ class MM_ObjectAllocationAPI
 		}
 
 		/* Initialize the object */
-		J9NonIndexableObject *objectHeader = (J9NonIndexableObject *) instance;
-		objectHeader->clazz = (j9objectclass_t)(UDATA)clazz;
-
-		if (initializeSlots) {
-			memset(objectHeader + 1, 0, dataSize);
-		}
-
-		if (initializeLockWord) {
-			*J9OBJECT_MONITOR_EA(currentThread, instance) = OBJECT_HEADER_LOCK_RESERVED;
+		if (J9VMTHREAD_COMPRESS_OBJECT_REFERENCES(currentThread)) {
+			J9NonIndexableObjectCompressed *objectHeader = (J9NonIndexableObjectCompressed*) instance;
+			objectHeader->clazz = (j9objectclass_t)(UDATA)clazz;
```

Comment:
similarly as in Indexable Allocation cast should be `(U_32)(UDATA)`

Reply:
Fixed.